scripts/dist-git/remove_unused_sources

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/bash
+
+if [[ $# != 3 ]]; then
+    echo "invalid number of arguments"
+    exit 1
+fi
+
+package="$1"
+lookaside_dir="$2"
+git_dir="$3"
+
+pkg_git_dir="$git_dir/$package.git"
+pkg_dir="$lookaside_dir/$package"
+
+if [ ! -d "$pkg_dir" ]; then
+    echo "$pkg_dir is not a valid directory."
+    exit 1
+fi
+if [ ! -d "$pkg_git_dir" ]; then
+    echo "$pkg_git_dir is not a valid directory."
+    exit 1
+fi
+
+pushd /tmp > /dev/null
+
+git clone "$pkg_git_dir" 2> /dev/null
+cloned_dir=`basename "$package"`
+
+if [ -d "$cloned_dir" ]; then
+    pushd "$cloned_dir" > /dev/null
+else
+    popd > /dev/null
+    exit 2
+fi
+
+branches=()
+whitelist=()
+
+eval "$(git for-each-ref --shell --format='branches+=(%(refname))' refs/remotes/)"
+for branch in "${branches[@]##*/}"; do
+    git checkout "$branch" &> /dev/null
+    if [ -f sources ]; then
+        while read line; do
+	    ll=($line)
+	    whitelist+=(${ll[1]})
+        done < sources
+    fi
+done
+popd > /dev/null
+
+for source_dir in "$pkg_dir"/*; do
+    srcdir="${source_dir##*/}"
+
+    if [[ "${whitelist[@]}" =~ "${srcdir}" ]]; then
+        continue
+    else
+    	rm -rf "$source_dir"
```

Comment:
I'd maybe do only `rm "$source_dir"`, at least at the beginning it is slightly safer variant ...